vllm_v1/tokenizer/detokenizer.py

```diff
@@ -0,0 +1,158 @@
+import multiprocessing
+
+from dataclasses import dataclass
+from typing import Dict, List, Optional, Tuple
+
+import msgspec
+import zmq
+
+from vllm.transformers_utils.tokenizer import get_tokenizer
+from .detokenizer_utils import (convert_prompt_ids_to_tokens,
+                                detokenize_incrementally)
+
+
+class RequestData(msgspec.Struct):
+
+    # [num_reqs]
+    request_ids: List[str]
+    prompt_token_ids: List[List[int]]
+    new_token_ids: List[List[int]]
+    skip_special_tokens: List[bool]
+    spaces_between_special_tokens: List[bool]
+
+    # [num_free_reqs]
+    free_request_ids: List[str]
+
+
+class DetokenizedData(msgspec.Struct):
+
+    # [num_reqs]
+    request_ids: List[str]
+    detokenized_texts: List[str]
+
+
+@dataclass
+class RequestState:
+
+    req_id: str
+
+    token_ids: List[int]
+    tokens: List[str]
+
+    prefix_offset: int
+    read_offset: int
+
+    skip_special_tokens: bool
+    spaces_between_special_tokens: bool
+
+    output_text: str = ""
+
+
+class Detokenizer(multiprocessing.Process):
```

Comment:
iiuc there is nothing preventing Detokenizer being a Ray actor but still use zmq for communication right? Is my understanding correct?

Reply:
Yes, you're right. I just used multiprocessing because the de-tokenizer will stay in the same machine as the engine.